File: app/src/main/java/com/example/saka/myapplication/adaper/AddFriendAdapter.java
package com.example.saka.myapplication.adaper;

import androidx.recyclerview.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;

import com.example.saka.myapplication.R;
import com.example.saka.myapplication.model.User;
import com.firebase.ui.database.FirebaseRecyclerAdapter;
import com.google.firebase.database.Query;

/**
 * Created by Saka on 04-Jun-17.
 */

public class AddFriendAdapter extends FirebaseRecyclerAdapter<User, AddFriendAdapter.AddFriendViewHolder> {

    private OnAddFriendItemClick recyclerItemClick;

    public AddFriendAdapter(Class<User> modelClass, int modelLayout, Class<AddFriendViewHolder> viewHolderClass, Query ref) {
        super(modelClass, modelLayout, viewHolderClass, ref);
    }

    public void setOnItemClickListener(OnAddFriendItemClick recyclerItemClick){
        this.recyclerItemClick = recyclerItemClick;
    }

    @Override
    public AddFriendViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(mModelLayout, parent, false);
        return new AddFriendViewHolder(view);
    }

    @Override
    protected void populateViewHolder(AddFriendViewHolder viewHolder, User model, int position) {
        viewHolder.name.setText(model.getName());
    }

    public class AddFriendViewHolder extends RecyclerView.ViewHolder {
        private ImageView imgProfile;
        private TextView name;

        private AddFriendViewHolder(View itemView) {
            super(itemView);
            imgProfile = (ImageView) itemView.findViewById(R.id.vh_add_friend_img);
            name = (TextView) itemView.findViewById(R.id.vh_add_friend_txt_name);
        }
    }

    public interface OnAddFriendItemClick{
        void onItemClick(View v, int position);
    }
}

File: app/src/main/java/com/blockdev/birdy/adaper/AddFriendAdapter.kt
package com.blockdev.birdy.adaper

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.blockdev.birdy.model.User
import com.blockdev.birdy.R
import com.blockdev.birdy.adaper.AddFriendAdapter.AddFriendViewHolder
import com.firebase.ui.database.FirebaseRecyclerAdapter
import com.google.firebase.database.Query

/**
 * Created by Saka on 04-Jun-17.
 */
class AddFriendAdapter(
    modelClass: Class<User?>?,
    modelLayout: Int,
    viewHolderClass: Class<AddFriendViewHolder?>?,
    ref: Query?,
) : FirebaseRecyclerAdapter<User, AddFriendViewHolder>(
    modelClass,
    modelLayout,
    viewHolderClass,
    ref,
) {
    private var recyclerItemClick: OnAddFriendItemClick? = null
    fun setOnItemClickListener(recyclerItemClick: OnAddFriendItemClick?) {
        this.recyclerItemClick = recyclerItemClick
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): AddFriendViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(mModelLayout, parent, false)
        return AddFriendViewHolder(view)
    }

    override fun populateViewHolder(viewHolder: AddFriendViewHolder, model: User, position: Int) {
        viewHolder.name.text = model.name
    }

    inner class AddFriendViewHolder constructor(itemView: View) :
        RecyclerView.ViewHolder(itemView) {
        private val imgProfile: ImageView
        val name: TextView

        init {
            imgProfile = itemView.findViewById<View>(R.id.vh_add_friend_img) as ImageView
            name = itemView.findViewById<View>(R.id.vh_add_friend_txt_name) as TextView
        }
    }

    interface OnAddFriendItemClick {
        fun onItemClick(v: View?, position: Int)
    }
}
